python/pyspark/ml/util.py

```diff
@@ -283,3 +289,124 @@ def numFeatures(self):
         Returns the number of features the model was trained on. If unknown, returns -1
         """
         return self._call_java("numFeatures")
+
+
+@inherit_doc
+class DefaultParamsWritable(MLWritable):
+
+    # overrides the write() function in MLWriteable
+    # users call .save() in MLWriteable which calls this write() function and then calls
+    # the .save() in DefaultParamsWriter
+    # this can be overridden to return a different Writer (ex. OneVsRestWriter as seen in Scala)
+    def write(self):
+        # instance of check for params?
+        return DefaultParamsWriter(self)
+
+
+@inherit_doc
+class DefaultParamsWriter(MLWriter):
+
+    def __init__(self, instance):
+        super(DefaultParamsWriter, self).__init__()
+        self.instance = instance
+        self.sc = SparkContext._active_spark_context
+
+    # if a model extends DefaultParamsWriteable this save() function is called
+    def save(self, path):
+        if self.shouldOverwrite:
+            # This command removes a file. Is this enough?
+            os.remove(path)
```

Comment:
We use `saveAsTextFile` to save metadata, it is possible to save to `HDFS` or other non-local FS, so here we cannot use `os.remove` to remove it. We need first get the File System of the `path` and call relative API.